os/fs/smartfs/smartfs_utils.c

```diff
@@ -1339,7 +1345,7 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 	entry->flags = new_entry.flags;
 
 	/* Now write the new entry to the parent directory sector */
-	if (parentdirsector != psector) {
+	if (last_parentdirsector != psector) {
```

Comment:
same comment in line 1300

Reply:
It is the same check.
Using alias variables for constant comparison as done in other fs layer functions.

Reply:
As I understand, some variable like a entry(mostly sf->entry) are added to prevent corruption by error which happened in the middle of function.
But It seems that dsector(psector) & last_dsector(last_parentdirsector) is not changed here....

Reply:
I'd like to take this up in a separate set of changes then.
Depending on how all the functions use aliases and how they can be optimized.

Reply:
Let's let the flow follow the same general tradition currently.